core/ffmpeg/ffmpeg.go

```diff
@@ -45,10 +56,25 @@ func (e *ffmpeg) ExtractImage(ctx context.Context, path string) (io.ReadCloser,
 	if _, err := ffmpegCmd(); err != nil {
 		return nil, err
 	}
-	args := createFFmpegCommand(extractImageCmd, path, 0)
+
+	imgFormat, err := e.GetVideoStreamType(ctx, path)
```

Comment:
I'm not fan of increasing memory and cpu consumption to handle an edge case like this, calling ffmpeg twice for each image. Can't we cameup with a "smart" ffmpeg command to do that in one call?

Reply:
The first call will not involve any conversion, as it just asks `ffmpeg` to read & parse the metadata of the media file.

Alternatively, we can ask `ffmpeg` to extract the raw stream like before, check for known image magic header, then ask `ffmpeg` to convert the stream if we don't recognize it.

Reply:
Re-implemented the logic (integration test needed), asking ffmpeg to convert the file when we don't know what the buffer is.
Happy path - everything is fast. Sad path - ffmpeg is reading the audio file twice.